### DISCUSSION TITLE
Memory copy

### DISCUSSION THREAD

**Discussion by jacestk** [rallepalaveev/A2DVI#2] [General]
Just some questions arround the memory recopy... It should take lot of cycles of the bus to recopy the data? The Apple // is not slower (less time left for the cpu) because of this? I am a bit lost about it...

**Comment by oliverschmidt** [comment]
I guess that could be answered in several ways. One way: In order to conceptually(!) have any influence whatsoever on the speed of the Apple II, the card would need to make use either of DMA or RDY. But the relevant pins aren't connected in the first place. So even a
"stupid", bogus or malicious A2DVI firmware would have no way to influence the speed of the Apple II.
Another way: Apple II graphics cards like the A2DVI only listen passively to the data on the Apple II bus. They don't take any action whatsoever from the Apple II perspective.
Yet another way: They copy process you likely refer to is performed by the micro controller on the card. The 6502 doesn't play any role.

**Comment by rallepalaveev** [comment]
Thanks Oliver for addressing this. In simple words, the A2DVI card only eavsdrops on the Apple2 and builds a picture of what's going on and what should be on the display. It monitors all traffic on the address and data busses as well as some control signals to achieve this. However, it does not interfere in any way with what's going on in the computer.

**Reply by jacestk** [reply]
Ok I understand now. Yep it is a really simple way to monitor in fact the address and datas lines! I just didnt thought about it 😅
